enhancements/network/configure-ovs-alternative.md

```diff
@@ -0,0 +1,498 @@
+          - name: br-ex
+      - name: br-ex
+        type: ovs-interface
+        state: up
+        ipv4:
+          enabled: true
+          dhcp: true
+        ipv6:
+          enabled: false
+          dhcp: false
+
+The name field will need to be something that uniquely identifies a given host.
+I'm unsure if name is necessarily the best way to do that, but it is one valid
+option. I reused the networkConfig name from the baremetal feature for
+consistency. Perhaps something different would be preferable to avoid confusion
+though?
+
+Note that although the configuration can be provided on a per-host basis, it is
+not mandatory to do so. YAML anchors can be used to duplicate a single
+configuration across multiple nodes. For example:
+
+networking:
+  networkType: OVNKubernetes
+  machineNetwork:
+  - cidr: 192.0.2.0/24
+  [...]
+  hostConfig:
+  - name: master-0
+    networkConfig: &BOND
+      interfaces:
+      - name: br-ex
+        ...etc...
+  - name: master-1
+    networkConfig: *BOND
+  - name: master-2
+    networkConfig: *BOND
+
+This reduces duplication in install-config and the chance of errors due to
+unintended differences in node config.
+
+When this mechanism is in use, a flag will be set to indicate to configure-ovs
+that it should not attempt to configure br-ex.
+
+If changes to the bridge configuration are desired on day 2 (e.g. adding a VLAN
+or changing MTU), the user will need to install Kubernetes-NMState and use it
+to apply those changes directly to br-ex.
+
+#### Variation [optional]
+
+This could be implemented almost entirely with machine-configs, with only a simple
+service on the host to apply the provided NMState files. This is the approach
+the [prototype implementation](https://docs.google.com/document/d/1Zp1J2HbVvu-v4mHc9M594JLqt6UwdKoix8xkeSzVL98/edit#heading=h.fba4j9nvp0nl)
+used since per-host configuration was not available any other way. However,
+this feels messy as it requires every node to have the configuration for every
+other node in the cluster and select the correct one for itself. It does have
+the benefit of requiring significantly less work to implement. If users are
+writing their NMState configs in machine-configs instead of via a top-level
+OpenShift API perhaps this also side-steps the concerns about API style
+compatibility between OpenShift and NMState?
+
+### API Extensions
+
+This will not directly modify the API of OpenShift. The API for this feature is
+implemented in Kubernetes-NMState, and this does not require any specific
+modifications in that project at this time.
+
+### Implementation Details/Notes/Constraints [optional]
+
+What are the caveats to the implementation? What are some important details that
+didn't come across above. Go in to as much detail as necessary here. This might
+be a good place to talk about core concepts and how they relate.
+
+### Risks and Mitigations
+
+Because this gives the deployer a great deal of control over the bridge
```

Comment:
On the positive side, should the deployer apply a configuration that renders the node inaccessible, the trio (NetworkManager, nmstate and kubernetes-nmstate) are expected to recover by rolling back to the previous network config state.

Reply:
Good point. I'll mention that as a mitigation to the risk.